src/coreclr/src/jit/assertionprop.cpp

```diff
@@ -1625,7 +1625,11 @@ void Compiler::optDebugCheckAssertion(AssertionDsc* assertion)
                     assert(assertion->op2.u1.iconFlags != 0);
                     break;
                 case O1K_LCLVAR:
-                    assert((lvaTable[assertion->op1.lcl.lclNum].lvType != TYP_REF) || (assertion->op2.u1.iconVal == 0));
+                    if (!IsTargetAbi(CORINFO_CORERT_ABI)) // CoreRT allows non-zero const refs for frozen strings.
```

Comment:
The frozen strings are not really inherent part of CoreRT ABI. Nothing prevents them from being used with CoreCLR as well, we just happen to not do that currently.

It would be better to have a `bool frozenObjectsUsed;` flag that gets set whenever the JIT/EE interface returns a frozen object, and use this flag for these asserts.

It would be better to have a

Reply:
good idea, thank you, fixed.